components/OpenIbox.js

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import StyledButton from './StyledButton';
+
+function OpenEmailProviderButton({ email }) {
+  const gmailregex = /@gmail\.com$/.test(email);
+  const hotmailregex = /@outlook\.com$/.test(email);
+  const gmaillink =
+    'https://mail.google.com/mail/u/2/#advanced-search/subject=Open+Collective%3A+Login&amp;subset=all&amp;within=1d';
+  const hotmaillink = 'https://outlook.live.com/mail/inbox';
+  const [btntext] = useState(gmailregex ? 'Gmail' : hotmailregex ? 'HotMail' : 'Inbox');
+  const [link] = useState(gmailregex ? gmaillink : hotmailregex ? hotmaillink : '');
+  const [emailProvider] = useState(gmailregex || hotmailregex ? true : false);
```

Comment:
Do we really need to use a React state for these?

Reply:
Thanks for the feedback, I have removed state and it still works perfect.